cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -1424,7 +1434,11 @@ func generateMountsModeRun(mst *initramfsMountsState) error {
 
 	// TODO: do we actually need fsck if we are mounting a mapper device?
 	// probably not?
-	if err := doSystemdMount(unlockRes.FsDevice, boot.InitramfsDataDir, fsckSystemdOpts); err != nil {
+	dataMountOpts := &systemdMountOptions{
```

Comment:
```
// fsck and mount with nosuid to prevent snaps from being able 
// to bypass the sandbox by creating suid root files  there and 
// trying to escape the sandbox
```